api/mesh/v1alpha1/timeout_helpers.go

```diff
@@ -12,3 +17,22 @@ func (x *Timeout_Conf) GetConnectTimeoutOrDefault(defaultConnectTimeout time.Dur
 	}
 	return connectTimeout.AsDuration()
 }
+
+func (m *Timeout) UnmarshalJSON(data []byte) error {
+	return proto_util.FromJSON(data, m)
+}
+
+func (m *Timeout) MarshalJSON() ([]byte, error) {
+	return proto_util.ToJSON(m)
+}
+func (t *Timeout) DeepCopyInto(out *Timeout) {
+	proto.Merge(out, t)
+}
+func (t *Timeout) DeepCopy() *Timeout {
+	if t == nil {
+		return nil
+	}
+	out := new(Timeout)
+	t.DeepCopyInto(out)
+	return out
+}
```

Comment:
Can't these all be auto generated too? Sounds like it's a bunch of boilerplate code that could just autogenerated in a file.

Reply:
@lahabana They can. This is what I was referring to in the summary with "There is probably room for more generation in api/mesh/v1alpha1/". These were really easy to append to the existing files, but we can put them under automation if that is what we want to spend our time on.

Reply:
I guess now that it's done it's ok. It's just easier to generate new policies if everything is auto gen :)

Reply:
Will something fail to compile somewhere if someone forgets to add the custom Un/MarshalJSON methods?

Reply:
Since this PR is a high priority for the next release which will be cut soon, I propose we move this code under auto-gen as part of this PR only if there is time after fixing all the tests and other necessary fallout from this change.

Reply:
Agreed with @parkanzky (it's ok to open an issue to do this after if no time on the PR).
@michaelbeaumont yes it will though it's annoying to have to fight against the compiler when creating things no?

Reply:
Mentioned above, will open a subsequent PR with template suggestions and open an issue if I don't get to that today.